src/AbstractServiceProvider.php

```diff
@@ -70,4 +74,77 @@ protected function makeClient(Application $app)
 
         return $client;
     }
+
+    /**
+     * Setup user detection.
+     *
+     * @param \Bugsnag\Client    $client
+     * @param \Silex\Application $app
+     *
+     * @return void
+     */
+    protected function setupUserDetection(Client $client, Application $app)
+    {
+        try {
+            $tokens = $app['security.token_storage'];
+            $checker = $app['security.authorization_checker'];
+        } catch (InvalidArgumentException $e) {
+            return;
+        }
+
+        $client->registerCallback(new CustomUser(function () use ($tokens, $checker) {
+            $token = $tokens->getToken();
+
+            if (!$token || !$checker->isGranted('IS_AUTHENTICATED_REMEMBERED')) {
+                return;
+            }
+
+            $user = $token->getUser();
+
+            if ($user instanceof UserInterface) {
+                return ['id' => $user->getUsername()];
+            }
+
+            return ['id' => (string) $user];
+        }));
+    }
+
+    /**
+     * Setup the client paths.
+     *
+     * @param \Bugsnag\Client $client
+     * @param string|null     $strip
+     * @param string|null     $project
+     *
+     * @return void
+     */
+    protected function setupPaths(Client $client, $strip, $project)
+    {
+        if ($strip) {
+            $client->setStripPath($strip);
+
+            if (!$project) {
+                $client->setProjectRoot("{$strip}/src");
+            }
+
+            return;
+        }
+
+        if ($project) {
+            if ($base && substr($project, 0, strlen($base)) === $base) {
+                $client->setStripPath($base);
+            }
+
+            $client->setProjectRoot($project);
+
+            return;
+        }
+
+        $base = realpath(__DIR__.'/../../../../');
```

Comment:
Shouldn’t there be directory separators here instead of concatenating forward slashes?